docs/building.md

```diff
@@ -0,0 +1,167 @@
+  [testing with TPM emulator](/tpm-simulators/), or to limit Fobnail usage to
+  one specific TPM vendor. Either this or `FOBNAIL_EK_ROOT_DIR` (or both)
+  **must be specified**.
+
+#### Building and flashing firmware to Fobnail Token
+
+For hardware setup instructions see [Flashing preparation](flashing_preparation.md).
+
+Building and flashing is performed by executing (from `fobnail` directory):
+
+```shell
+env FOBNAIL_PO_ROOT=root_ca.crt FOBNAIL_EK_ROOT_DIR=tpm_ek_roots \
+    ./build.sh -t nrf --run
+```
+
+A console with Fobnail Token output will be displayed. It is not required for
+normal operation but can be useful for debugging. It can be closed with Ctrl-C
+at any point. After that, Token can be used without nRF52840-DK - just plug it
+wherever it's needed.
+
+#### Building and running firmware locally
+
+Make sure you have the network set up properly. See [Networking setup](networking_setup.md)
+for instructions. For firmware to work properly you need the `fobnail0` network
+interface configured.
+
+To build and run firmware execute the following commands (from `fobnail`
+directory).
+
+```shell
+env FOBNAIL_LOG=info FOBNAIL_PO_ROOT=root_ca.crt \
+    FOBNAIL_EK_ROOT_DIR=tpm_ek_roots ./build.sh -t pc --run
+```
+
+`FOBNAIL_LOG` environment variable sets log level. The possible log levels are:
+`error`, `warning`, `info`, `debug`, `trace`. This variable is valid only for PC
+target.
+
+Another variable used only on PC is `FOBNAIL_DEVICE_ID` - on hardware we used
+FICR registers to create a device ID, this gives a way of configuring it for
+emulation. This variable is optional, without it an ID of 0 is used.
+
+`build.sh` automatically starts Token emulation. It runs until it's terminated
+with Ctrl-C.
+
+## PC applications
+
+Platform Owner and Attester have similar set of prerequisites, listed below.
+Depending on use case, they may or may not be run on the same PC, so these steps
+may have to be repeated on different computers.
+
+```shell
+sudo apt -y install git make gcc autoconf automake pkg-config libtool libssl-dev
+```
+
+Both applications use [libcoap with v3 API](https://github.com/obgm/libcoap/tree/release-4.3.0)
+which isn't provided by main Linux distributions yet, so it has to be built from
+sources and installed:
+
+```shell
+git clone https://github.com/obgm/libcoap.git --recurse-submodules
+cd libcoap
+git checkout release-4.3.0
+./autogen.sh && \
+./configure --exec-prefix=/usr --disable-tests --disable-documentation \
+    --disable-manpages --enable-dtls --with-tinydtls --enable-fast-install && \
+make && sudo make install
+```
+
+#### Platform Owner
+
+With prerequisites installed, building Platform Owner application is simple -
+just clone and build:
+
+```shell
+git clone https://github.com/fobnail/fobnail-platform-owner --recurse-submodules
+cd fobnail-platform-owner
+make
+```
+
+Produced binary is located in `bin/fobnail-platform-owner`, from where it can be
+moved to `PATH` or just started from there.
```

Comment:
I would expect example usage here. I managed to run provisioning, I assume everything is fine:

```
(venv) [22:07:53] pietrushnic:fobnail git:(main*) $ env FOBNAIL_LOG=info FOBNAIL_PO_ROOT=root_ca.crt \
    FOBNAIL_EK_ROOT_DIR=tpm_ek_roots ./build.sh -t pc --run
useradd: warning: the home directory already exists.
Not copying any file from skel directory into it.
    Finished dev [optimized + debuginfo] target(s) in 0.21s
     Running `target/x86_64-unknown-linux-gnu/debug/fobnail`
 INFO  fobnail > Hello from main
 INFO  pal_pc::led > LED controller state: TokenNotProvisioned
 INFO  fobnail     > new client: 169.254.0.8:40625
 INFO  fobnail::server::token_provisioning > Commencing token provisioning
 INFO  fobnail::server::token_provisioning > Certificate chain loaded
 INFO  fobnail::server::token_provisioning > Generating new Ed25519 keypair
 INFO  fobnail::server::token_provisioning > Token provisioning complete
 INFO  pal_pc::led                         > LED controller state: TokenProvisioningComplete
 INFO  pal_pc::led                         > LED controller state: TokenWaiting
 INFO  fobnail                             > disconnect client: 169.254.0.8:40625
```

But how would regular users know?

Also, it would be great if the naming convention between `fobanil-platform-owner` parameters would be consistent with file names used in TL;DR and cert generation documentation.

Reply:
Names changed in https://github.com/fobnail/docs/pull/47/commits/ee9239f5f3f4e4d6d42c21c0cd2f9bf2d615f076